pulp_container/app/tasks/sync_stages.py

```diff
@@ -61,40 +60,41 @@ def __init__(self, remote, signed_only):
         self.manifest_dcs = []
         self.signature_dcs = []
 
-    async def _download_and_save_artifact_data(self, manifest_url):
+    async def _download_artifact_data(self, manifest_url):
         downloader = self.remote.get_downloader(url=manifest_url)
         response = await downloader.run(extra_data={"headers": V2_ACCEPT_HEADERS})
         with open(response.path, "rb") as content_file:
-            raw_data = content_file.read()
+            raw_bytes_data = content_file.read()
         response.artifact_attributes["file"] = response.path
 
-        saved_artifact = await save_artifact(response.artifact_attributes)
-        content_data = json.loads(raw_data)
+        raw_text_data = raw_bytes_data.decode("utf-8")
+        content_data = json.loads(raw_bytes_data)
 
-        return saved_artifact, content_data, raw_data, response
+        return content_data, raw_text_data, response
 
     async def _check_for_existing_manifest(self, download_tag):
         response = await download_tag
 
         digest = response.headers.get("docker-content-digest")
 
-        if digest and (
-            manifest := await Manifest.objects.prefetch_related("contentartifact_set")
-            .filter(digest=digest, _artifacts__isnull=False)
-            .afirst()
-        ):
-            saved_artifact = await manifest._artifacts.aget()
-            content_data, raw_data = await sync_to_async(get_content_data)(saved_artifact)
+        if digest and (manifest := await Manifest.objects.filter(digest=digest).afirst()):
+            content_data = json.loads(manifest.data)
```

Comment:
This is going to fail for manifests that have artifacts still in place and the users triggered a sync task. We need to check if `manifest.data` is populated before loading the data.

Correct me if I am wrong, but this implementation will fail on a system which contains already synced manifests and an admin has not run the django-management command yet.